src/components/WalletDropdown/MiniPortfolio/PortfolioLogo.tsx

```diff
@@ -147,13 +137,11 @@ export function PortfolioLogo({
   }
 
   const L2Logo =
-    chainId === SupportedChainId.MAINNET ? null : (
-      <div>
-        {chainLogo && <ChainLogoSquareBackground />}
-        {!squareLogoUrl && logoUrl && <SquareBackgroundForNonSquareLogo />}
-        {chainLogo && <StyledChainLogo isSquare={!!squareLogoUrl} src={chainLogo} alt="chainLogo" />}
-      </div>
-    )
+    chainId !== SupportedChainId.MAINNET && chainLogo ? (
+      <L2LogoContainer $backgroundColor={squareLogoUrl ? theme.backgroundSurface : theme.textPrimary}>
+        <StyledChainLogo $fillParent={Boolean(squareLogoUrl)} src={chainLogo} alt="chainLogo" />
+      </L2LogoContainer>
```

Comment:
honestly might be cleaner to just do two separate components here instead of trying to generalize for both square and non-square